### PR TITLE
fix: [CDS-82105]: added workingDir for scripts

### PR DESCRIPTION
**Pull request by vitsafronovici**
## Describe your changes
Location for WinRm should point to the one defined in workingDir field for scripts, currently the location is pointing to %USERPROFILE%

Before:
image

After:
image

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54504)

### CODE REVIEW

**Comment by vitsafronovici** [comment]
trigger uts

**Comment by vitsafronovici** [comment]
trigger messagecheck

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/winrm/WinRmUtils.java [hunk 1]
```diff
@@ -135,4 +135,10 @@ public static CommandExecutionStatus getStatus(ExecuteCommandResponse executeCom
   public static String getWorkingDir(String workingDirectory) {
     return isNotEmpty(workingDirectory) ? workingDirectory : WINDOWS_HOME_DIR;
   }
+
+  public static String getWorkingDir(NgCommandUnit commandUnit) {
```
Comment (acasian-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/winrm/WinRmUtils.java:139]: destination path is only relevant for copy/download artifact, correct? i'm not following the complete use case for now, can you explain the problem?

Comment (vitsafronovici) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/winrm/WinRmUtils.java:139]: Before my changes the workingDir for Script was null and it was set to %USERPROFILE%.
If you follow  the bug description, the customer provided the workingDir for script, and expected that Get-Location to point to defined directory. The customer was using running some script in this manner:
Get-Location & .\installapp.cmd

**Comment by vitsafronovici** [comment]
trigger messagecheck

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/winrm/WinRmUtils.java (hunk 1 above)
Comment (acasian-harness) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/winrm/WinRmUtils.java:139]: got it, we should have the same for ssh in this case

**Review by acasian-harness — APPROVED** [review]

**Review by ivan-83 — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54504#-:-NhRqu4dEfPzPAD0r1gU:bnfp4nl)* status: 2 file(s) to review.

**Comment by vitsafronovici** [comment]
trigger review

**Pull request merged**